src/state_point.F90

```diff
@@ -126,151 +147,197 @@ subroutine write_state_point()
-                 group="tallies/tally" // trim(to_str(i)) // "/moments")
-            k = k + 1
-          case (SCORE_SCATTER_PN, SCORE_NU_SCATTER_PN)
-            do n_order = 0, t % moment_order(k)
-              moment_name = 'P' // trim(to_str(n_order))
+      if (n_tallies > 0) then
+
+        ! Print list of tally IDs
+        allocate(id_array(n_tallies))
+        allocate(key_array(n_tallies))
+
+        ! Write all tally information except results
+        do i = 1, n_tallies
+          tally => tallies(i)
+          key_array(i) = tally % id
+          id_array(i) = i
+        end do
+
+        call sp % write_data(id_array, "ids", &
+             group="tallies", length=n_tallies)
+        call sp % write_data(key_array, "keys", &
+             group="tallies", length=n_tallies)
+
+        deallocate(key_array)
+
+        ! Write all tally information except results
+        TALLY_METADATA: do i = 1, n_tallies
+
+          ! Get pointer to tally
+          tally => tallies(i)
+
+          call sp % write_data(len(tally % label), "label_size", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+          if (len(tally % label) > 0) then
+            call sp % write_data(tally % label, "label", &
+                 group="tallies/tally " // trim(to_str(tally % id)))
+          endif
+
+          call sp % write_data(tally % estimator, "estimator", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+          call sp % write_data(tally % n_realizations, "n_realizations", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+          call sp % write_data(tally % n_filters, "n_filters", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+
+          ! Write filter information
+          FILTER_LOOP: do j = 1, tally % n_filters
+
+            call sp % write_data(tally % filters(j) % type, "type", &
+                 group="tallies/tally " // trim(to_str(tally % id)) // &
+                 "/filter " // to_str(j))
+            call sp % write_data(tally % filters(j) % n_bins, "n_bins", &
+                 group="tallies/tally " // trim(to_str(tally % id)) // &
+                 "/filter " // to_str(j))
+            if (tally % filters(j) % type == FILTER_ENERGYIN .or. &
+                tally % filters(j) % type == FILTER_ENERGYOUT) then
+              call sp % write_data(tally % filters(j) % real_bins, "bins", &
+                   group="tallies/tally " // trim(to_str(tally % id)) // &
+                   "/filter " // to_str(j), &
+                   length=size(tally % filters(j) % real_bins))
+            else
+              call sp % write_data(tally % filters(j) % int_bins, "bins", &
+                   group="tallies/tally " // trim(to_str(tally % id)) // &
+                   "/filter " // to_str(j), &
+                   length=size(tally % filters(j) % int_bins))
+            end if
```

Comment:
Thanks for pointing this out @samuelshaner. This issue is *very* similar to many of the other issues I tried to resolve with the refactored ``openmc.statepoint`` Python API in this PR - namely, the use of indices vs. ids for tallies, nuclides, etc. Of course, the user would much prefer to see intuitive IDs used to identify various objects in the statepoint files rather than the indices which OpenMC uses to internally categorize these objects. 

You'll notice that the way I deal with this is to export the internal IDs for these objects to the statepoint files, but allow the user to switch to IDs with the ``openmc.statepoint.link_with_summary(...)`` routine. This parses the ``summary.h5`` file and links the indices with IDs. Since the meshes are not printed out to the summary files (see ``hdf5_summary.F90``), I did not include them in this routine. I think your fix is a good approach, though I'd like to hear @paulromano's opinion.

@samuelshaner - if you've already tried this out, do you want to submit a PR to my fork?